cluster/region/region.go

```diff
@@ -93,7 +94,21 @@ func (r *region) TransferLeader(peer string) error {
 }
 
 func (r *region) AddPeer(peer string) error {
-	panic("implement me")
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	contains := func(arr []string, it string) bool {
+		for _, n := range arr {
+			if n == it {
+				return true
+			}
+		}
+		return false
+	}
+	if !contains(r.peers, peer) {
+		r.peers = append(r.peers, peer)
```

Comment:
This may not be a simple array expansion, but a raft cluster expansion where you need to call raft's methods in the structure to add nodes. Like `r.raft.AddVoter()` to add peer into `raftgroup`.

Reply:
So, in our case, ServerID and ServerIP address are the same as `peer,` or are they stored elsewhere?